webapp/app/dataExport/components/resultsTable/resultsTable.js

```diff
@@ -15,24 +55,45 @@ const ResultsTable = (props) => {
   const i18n = useI18n()
   const tableRef = useRef(null)
   const [exportDisabled, setExportDisabled] = useState(true)
+  const [showTableExport, setShowTableExport] = useState(false)
+
+  // Unit consts
+  const tableSpec = SectionSpecs.getTableSpecExport(assessmentType, section)
+  const baseUnit = tableSpec.unit
+  const [unit, setUnit] = useState(baseUnit)
+
+  const onUnitChange = (value) => {
+    setShowTableExport(true)
+    setUnit(value)
+  }
+
+  useLayoutEffect(() => {
+    setShowTableExport(false)
+  }, [unit])
 
   useLayoutEffect(() => {
     setExportDisabled(resultsLoading)
   }, [resultsLoading])
 
   return (
     <div className="results-table">
-      <ButtonTableExport tableRef={tableRef} filename={`${assessmentType}-${section}`} disabled={exportDisabled} />
+      {!showTableExport && tableRef.current != null && (
```

Comment:
do not hide the button. setting exportDisabled to false and then true should be enough